File: src/lib/crypto/jwt.ts
import type { JWTPayload } from '../types';

/**
 * Decode JWT token payload without verification
 */
export function decodeJWTPayload(token: string): JWTPayload {
	const parts = token.split('.');
	if (parts.length !== 3) {
		throw new Error('Invalid JWT format');
	}

	// Add padding if needed
	let payloadB64 = parts[1];
	const padding = 4 - (payloadB64.length % 4);
	if (padding !== 4) {
		payloadB64 += '='.repeat(padding);
	}

	const payloadJson = atob(payloadB64);
	return JSON.parse(payloadJson);
}

File: ios/App/Accessy/Crypto/JWT.swift
import Foundation

/// JWT decoding utilities
class JWT {
    /// Decode JWT payload without verification (server already verified)
    static func decodePayload(_ token: String) throws -> JWTPayload {
        let parts = token.split(separator: ".")
        guard parts.count >= 2 else {
            throw CryptoError.invalidKeyFormat
        }

        let payloadPart = String(parts[1])
        guard let data = Encryption.safeB64Decode(payloadPart) else {
            throw CryptoError.invalidKeyFormat
        }

        let decoder = JSONDecoder()
        return try decoder.decode(JWTPayload.self, from: data)
    }
}
